### PR TITLE
argc: 1.14.0 -> 1.17.0

### PR DESCRIPTION
**Pull request by sdedovic**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sdedovic** [comment]
I have these changes overlayed for one of my projects and it works.

File: pkgs/development/tools/argc/default.nix [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
     rev = "v${version}";
-    hash = "sha256-Li/K5/SLG6JuoRJDz2DQoj1Oi9LQgZWHNvtZ1HVbj88=";
+    hash = "sha256-OvWXlb1LfUDLOAwAD/LAxoR8Wg8CAgWPXg6d76X/CJI=";
   };
 
-  cargoHash = "sha256-D1T9FWTvwKtAYoqFlR2OmLRLGWhPJ9D8J7lq/QKcBoM=";
+  cargoHash = "sha256-8NbVVSu2y8jgx6z8K4pjKidnRY74DQtYB6ueW7oIPx0=";
 
-  nativeBuildInputs = [ installShellFiles ];
+  nativeBuildInputs = [installShellFiles];
```
Comment (jonringer) [inline pkgs/development/tools/argc/default.nix:21]:
```suggestion
  nativeBuildInputs = [ installShellFiles ];
```

Comment (jonringer) [reply pkgs/development/tools/argc/default.nix:21]: Please use nixpkgs-fmt or `nixfmt` (which should be available) https://github.com/NixOS/rfcs/pull/166

Comment (sdedovic) [reply pkgs/development/tools/argc/default.nix]: Ah I believe I used https://github.com/kamadorueda/alejandra instead. I didn't realize the formatting RFC existed (and was accepted)

**Comment by ShamrockLee** [comment]
It would be great to bump to argc 1.17.0.

**Review by ShamrockLee — CHANGES_REQUESTED** [review]
argc: 1.15.0 -> 1.17.0

The three suggestions can be added together as a batch and committed using the web interface with the above commit message header. The update builds successfully on my x86_64-linux NixOS laptop.

**Comment by sdedovic** [comment]
> It would be great to bump to argc 1.17.0. I'm looking forward to its merge!

> The three suggestions can be added together as a batch and committed using the web interface with the above commit message header. The update builds successfully on my x86_64-linux NixOS laptop.

wow didn't even realize there are two new versions. updated the PR, thanks for the help!

**Review by ShamrockLee — APPROVED** [review]
LGTM code-wise.

I'm unable to run `nixpkgs-review` on my current setup (a borrowed laptop), but at least it builds on `x86_64-linux`.

The functionality tests seems to cover under the checkPhase from upstream. If so, a successful native build should guarantee that the result binary functions as expected.

**Comment by sdedovic** [comment]
It is now on version 1.19

I have lost interest - I have an overlay for my uses and while I wish to contribute, I am discouraged due to very slow review and merge timelines.

**Comment by ShamrockLee** [comment]
> I am discouraged due to the very slow review and merge timelines.

The slow review process is a long-lasting problem for Nixpkgs and is undoubtedly frustrating. The root cause is the vast rate of pull requests popping out and the comparatively small number of committers.

NixOS/rfcs#172 proposes a solution to this problem by allowing bot-based merging when a given number of maintainers approve a PR. This RFC is currently being discussed, as it changes the security model of Nixpkg's review process. You may be interested.

**Pull request closed without merging**